addons/stock/models/stock_quant.py

```diff
@@ -96,39 +94,36 @@ def _domain_product_id(self):
     tracking = fields.Selection(related='product_id.tracking', readonly=True)
     on_hand = fields.Boolean('On Hand', store=False, search='_search_on_hand')
 
+    # Inventory Fields
+    inventory_quantity = fields.Float(
+        'Inventoried Quantity', digits='Product Unit of Measure', groups='stock.group_stock_manager',
+        help="The product's counted quantity.")
+    inventory_diff_quantity = fields.Float(
+        'Difference', compute='_compute_inventory_diff_quantity', store=True,
+        help="Indicates the gap between the product's theoretical quantity and its counted quantity.",
+        readonly=True, digits='Product Unit of Measure', groups='stock.group_stock_manager')
+    inventory_date = fields.Date(
+        'Inventory Date', compute='_compute_inventory_date', store=True, readonly=False,
+        help="Next date the On Hand Quantity should be counted.")
+    user_id = fields.Many2one(
+        'res.users', 'User', groups='stock.group_stock_manager', help="User assigned to do product count.")
+
     @api.depends('quantity', 'reserved_quantity')
     def _compute_available_quantity(self):
         for quant in self:
             quant.available_quantity = quant.quantity - quant.reserved_quantity
 
-    @api.depends('quantity')
-    def _compute_inventory_quantity(self):
-        if not self._is_inventory_mode():
-            self.inventory_quantity = 0
-            return
-        for quant in self:
-            quant.inventory_quantity = quant.quantity
+    @api.depends('location_id')
+    def _compute_inventory_date(self):
+        for quants in self.filtered(lambda q: not q.inventory_date):
```

Comment:
I don't get it, if you loop in the `self.filtered(lambda q: not q.inventory_date)`:
- The `quants` contains always one record (rename into `quant`).
- Then mapped in location is useless, right (always one location_id)?
- Then the loop on location is useless ?

Then this code is strictly equals to 
```python
    @api.depends('location_id')
    def _compute_inventory_date(self):
        for quant in self.filtered(lambda q: not q.inventory_date):
             quant = quant.location_id._get_next_inventory_date()
```
Or I miss something ?

Reply:
Whoops, made a mistake. Fixed it so it's `quants = self.filtered(lambda q: not q.inventory_date)`. The idea is to batch the `inventory_date` assignment by location.